pinot-core/src/main/java/org/apache/pinot/core/util/trace/TraceContext.java

```diff
@@ -67,7 +68,7 @@ JsonNode toJson() {
 
     final String _traceId;
     final List<LogEntry> _logs = new ArrayList<>();
-    int _numChildren = 0;
+    AtomicInteger _numChildren = new AtomicInteger(0);
```

Comment:
Are you concerned with the case where multiple threads try to increment the `_numChildren`? Since we access Trace via `ThreadLocal<TraceEntry>`, it looks like we only access the trace object from a single thread. @Jackie-Jiang Can you confirm this?

Reply:
Reproduced in tests and updated them.